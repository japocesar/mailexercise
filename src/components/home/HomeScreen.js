import React, { useEffect, useState } from 'react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faStar, faExclamation } from "@fortawesome/free-solid-svg-icons";

import { getMails } from '../../selectors/getMails';

import './home.css';

export const HomeScreen = ({selectedTag, query}) => {

    const [mails, setMails] = useState([]);

    const handleSelect = (event) => {
        const el = event.currentTarget;
        const id = el.dataset.id;
        ( el.checked )
            ? document.querySelector(`[data-id="${id}"]`).classList.add('active')
            : document.querySelector(`[data-id="${id}"]`).classList.remove('active')
    }

    useEffect(() => {
        const mailsList = getMails(selectedTag);
        setMails( mailsList );
    }, [selectedTag]);

    useEffect(() => {
        let mailsList = [];
            mailsList = getMails(selectedTag, query);
        setMails( mailsList );
    }, [query]);

    return (
        <>
          <h1>{selectedTag}</h1>

          <ul className="list-group mb-5">
            { (!mails.length) &&
            <li>
                <h3 className="text-danger text-center">Not found emails for "{ query }"</h3>
            </li>
            }
            
            { mails.map( mail => 
            (
                <li key={ mail.id } data-id={ mail.id } className="list-group-item d-flex flex-wrap">
                    <div className="col-1 p-0">
                        <input onClick={ handleSelect } type="checkbox" data-id={ mail.id } aria-label="Checkbox for following text input"></input>
                    </div>
                    <div className="icons-container p-0 col-11 col-lg-2 d-flex">
                        <FontAwesomeIcon icon={faStar} />
                        <FontAwesomeIcon icon={faExclamation} />
                        <span>{ mail.sender }</span>
                    </div>
                    <div className="col-12 col-lg-8 p-0 pl-3 email-body d-flex flex-wrap">
                        <span className="subject content col-xs-12 col-md-auto pr-0"> { mail.subject } -</span>
                        <div className="col email-content pl-1" dangerouslySetInnerHTML={{ __html: mail.body }}></div>
                    </div>
                    <div className="p-0 text-right date-container"> { new Date(mail.date).toLocaleDateString() } </div>
                </li>
               
            ))
            }
        </ul>
        </>
    )
}
