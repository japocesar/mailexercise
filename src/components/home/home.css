h1 {
    text-transform: capitalize;
}

li {
    max-height: 2.5em;
    overflow: hidden;
}


.subject {
    width: fit-content;
    font-weight: bold;
}

.email-body p {
    overflow: hidden;
    text-overflow: ellipsis;
    white-space: nowrap;
    margin-top: -5px;
}

.email-content {
    overflow: hidden;
    padding-top: 5px;
}

svg {
    margin-right: 5px;
    color: lightgrey;
}

/* Extra small devices (portrait phones, less than 576px) */
@media (max-width: 575.98px) {
    .subject {
        width: 100%;
        overflow: hidden;
        text-overflow: ellipsis;
        white-space: nowrap;
    }

    li {
        max-height: 5.5em;
    }

    .date-container, .icons-container svg {
        display: none;
    }
}

/* Small devices (landscape phones, less than 768px) */
@media (min-width: 576px) and (max-width: 767.98px) {
    li {
        max-height: 4.5em;
    }
    .date-container, .icons-container svg {
        display: none;
    }
}