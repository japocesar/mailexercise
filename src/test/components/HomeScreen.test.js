import { shallow } from 'enzyme';
import React from 'react'
import { HomeScreen } from '../../components/home/HomeScreen';

describe('Tests in <HomeScreen />', () => {

    test('should display the component', () => {
        const wrapper = shallow(<HomeScreen selectedTag="all" />);
        expect( wrapper ).toMatchSnapshot();
    })
    

})