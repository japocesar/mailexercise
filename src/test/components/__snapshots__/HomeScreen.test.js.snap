// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in <HomeScreen /> should display the component 1`] = `
<Fragment>
  <h1>
    all
  </h1>
  <ul
    className="list-group mb-5"
  >
    <li>
      <h3
        className="text-danger text-center"
      >
        Not found emails for "
        "
      </h3>
    </li>
  </ul>
</Fragment>
`;
